File: practices/interview-bit/two-pointers/array-3-pointers.cpp
int Solution::minimize(const vector<int> &A, const vector<int> &B, const vector<int> &C) {
    int sizeA = (int)A.size();
    int sizeB = (int)B.size();
    int sizeC = (int)C.size();
    int i = 0, j = 0, k = 0;
    int ans = INT_MAX;
    while(i < sizeA && j < sizeB && k < sizeC) {
        int mn = min(A[i], min(B[j], C[k]));
        int mx = max(A[i], max(B[j], C[k]));
        ans = min(ans, mx - mn);
        
        if (mn == A[i]) i++;
        else if (mn == B[j]) j++;
        else k++;
    }
    return ans;
}

File: practices/interview-bit/two-pointers/array-3-pointers.py
class Solution:
    #@paramA: tuple of integers
    #@paramB: tuple of integers
    #@paramC: tuple of integers
    def minimize(self, A, B, C):
        sizeA = len(A)
        sizeB = len(B)
        sizeC = len(C)
        i = 0
        j = 0
        k = 0
        ans = float("inf")
        while(i < sizeA and j < sizeB and k < sizeC):
            mn = min(A[i], B[j], C[k])
            mx = max(A[i], B[j], C[k])
            ans = min(ans, mx - mn)

            if A[i] == mn:
                i += 1
            elif B[j] == mn:
                j += 1
            else:
                k += 1
        return ans
